.github/workflows/deploy.yml

```diff
@@ -1,68 +1,205 @@
-#      #    task-definition: ${{ env.task_definition }}
-#      #    container-name: ${{ env.container_name }}
-#      #    image: ${{ steps.build-image.outputs.image }}
-
-#      #- name: Deploy Amazon ECS task definition
-#      #  uses: aws-actions/amazon-ecs-deploy-task-definition@v1
-#      #  with:
-#      #    task-definition: ${{ steps.task-def.outputs.task-definition }}
-#      #    service: ${{ env.service_name }}
-#      #    cluster: ${{ env.cluster_name }}
-#      #    wait-for-service-stability: true
+name: Deploy to Amazon ECS
+
+env:
+  aws_region: eu-west-2
+  task_definition_name: webservers
+  service_name: webservers
+  cluster_name: webservers
+
+  ecr_repository_rails: "webserver-rails"
+  dockerfile_rails: "docker/rails.Dockerfile"
+
+  ecr_repository_nginx: "webserver-nginx"
+  dockerfile_nginx: "docker/nginx.Dockerfile"
+
+  ecr_repository_anycable_go: "webserver-anycable-go"
+  dockerfile_anycable_go: "docker/anycable-go.Dockerfile"
+
+  # TODO: Move this to be a secret
+  s3_bucket: "exercism-assets-staging"
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  deploy:
+    name: Upload images to ECR, assets to S3, and defintion to ECS
+    runs-on: ubuntu-latest
+    steps:
+      ###############
+      ###############
+      # Setup Steps #
+      ###############
+      ###############
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - uses: satackey/action-docker-layer-caching@v0.0.5
+        continue-on-error: true
+        with:
+          # Change these to rest teh cache
+          key: docker-layer-caching-deploy-{hash}
+          restore-keys: docker-layer-caching-deploy-
+          concurrency: 30
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_DEPLOY_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_DEPLOY_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.aws_region }}
+
+      - name: Login to Amazon ECR
+        id: login_to_ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      #########################
+      #########################
+      # Build the Rails image #
+      #########################
+      #########################
+      - name: Build and Push Rails Docker
+        env:
+          ECR_REGISTRY: ${{ steps.login_to_ecr.outputs.registry }}
+          ECR_REPOSITORY: ${{ env.ecr_repository_rails }}
+          IMAGE_TAG: ${{ github.sha }}
+          DOCKERFILE: ${{ env.dockerfile_rails }}
+          IMAGE_NAME: "${{ steps.login_to_ecr.outputs.registry}}/${{ env.ecr_repository_rails }}:${{ github.sha }}"
+        run: |
+          # Build a docker container and push it to ECR
+          docker build -t $IMAGE_NAME -f $DOCKERFILE .
+          docker push $IMAGE_NAME
+
+          # Retag this as the latest
+          docker tag $IMAGE_NAME $ECR_REGISTRY/$ECR_REPOSITORY:latest
+          docker push $ECR_REGISTRY/$ECR_REPOSITORY:latest
```

Comment:
The reason for this is that if we override the task-definition in terraform, we can just use `latest` there, without having to do complex interpolation of the latest version. However, for general practice it's good use to use the actual shas so that we can avoid race-conditions etc. We therefore tag the same image twice with sha and with `latest`.